### PR TITLE
Prevent Katex $ symbol from working across multiple lines

### LINKED ISSUE #1050: Katex single $ syntax shouldn't apply to multiple lines
https://discourse.joplin.cozic.net/t/katex-parse-error/1124/10?u=laurent

### PR DESCRIPTION
**Pull request by CalebJohn**
fixes #1050 

The real bug here was with [markdown-it-katex](https://github.com/waylonflinn/markdown-it-katex/issues) which hasn't been updated in 3 years and has 5 open pull requests, so it's unlikely to be fixed on that end.

It should be possible to use markdown-it-texmath as our rendering as well, but when I attempted that it had some errors rendering which didn't seem worthwhile to deal with.
So basically everything stays the same, except the behaviour is more intuitive.

On another note, there was some auto updates to package-lock.json by npm, is it okay to ignore these? I can remove it from the pull request if necessary.

### CODE REVIEW

**Comment by laurent22** [comment]
Thanks, just need to test it to confirm. For package-lock, it's ok to include it if package.json has changed too (which it has, in this case).
